Update supported ops doc for child devices (#1542)
thin-edge/thin-edge.io: docs/src/tutorials/supported_operations.md

### `thin-edge.io` Supported Operations API

The Supported Operations utilises the file system to add and remove operations. A special file placed in `/etc/tedge/operations` directory will indicate that an operation is supported.

> The specification for the operation files is described in `thin-edge.io` specifications repository [src/supported-operations/README.md](https://github.com/thin-edge/thin-edge.io-specs/blob/main/src/supported-operations/README.md)

Reviewer: Unrelated to this PR, but surprised to see a link to the specs repo, for a feature that we've exposed to the end-user. This could be replaced with a local reference to the `Adding new custom operation` section in this document itself.

Reviewer 2: I was surprised too and created that issue https://github.com/thin-edge/thin-edge.io/issues/1547

Supported operations are declared in the cloud specific subdirectory of `/etc/tedge/operations` directory.

### `thin-edge.io child` Supported Operations

When a new child device is added, the boot-straping of child device will create `/etc/tedge/operations/c8y/<child-device>` directory to store the supported operations by that particular child device.
It creates a file per Supported Operation. The operation can be dynamically added and removed. A special file placed in `/etc/tedge/operations/c8y/<child-device>` directory
will indicate that an operation is supported by that `child device`.

> 

Reviewer 2: I would remove this section or move it under "Custom Supported Operations".

I'm saying  that because this section is giving details of what is under `c8y` while cloud specific operations are introduced after.

Author: Moved below. Also, made it more generic by changing `/etc/tedge/operations/c8y/<child-device>` to `/etc/tedge/operations/<cloud-provider>/<child-device>`.

`thin-edge.io` stores the operations configuration files in the `/etc/tedge/operations/<cloud-provider>/` directory.

> ## `thin-edge.io child` Supported Operations

Reviewer: ```suggestion
## Supported Operations for Child Devices
```

Author: Fixed

> When a new child device is added, the boot-straping of child device will create `/etc/tedge/operations/<cloud-provider>/<child-device>` directory to store the supported operations by that particular child device.

Reviewer: ```suggestion
When a new child device is bootstrapped, it needs to create `/etc/tedge/operations/<cloud-provider>/<child-device>` directory to store the supported operations of that child device.
```

Author: Fixed

> It creates a file per Supported Operation. The operation can be dynamically added and removed. A special file placed in `/etc/tedge/operations/<cloud-provider>/<child-device>` directory
> will indicate that an operation is supported by that `child device`.

Reviewer: While writing markdown docs, it's a good practice to keep each sentence on its own line. Have re-ordered the sentences as well:
```suggestion
Every file placed in the `/etc/tedge/operations/<cloud-provider>/<child-device>` directory
indicates an operation supported by that child device.
The operation files can be dynamically added and removed.
```

Author: Fixed

`thin-edge.io` stores the operations configuration files in the `/etc/tedge/operations/<cloud-provider>/` directory.

> `thin-edge.io` stores the operations configuration files for the child devices in the `/etc/tedge/operations/<cloud-provider>/<child-device>` directory.

Reviewer 2: These two sentences are too repetitive. Make a single sentence that introduces the 2 levels.

You can move here the text related to child-device boot strap.

Author: moved child device-related things into the bootstrap part.

Now the new operation will be automatically added to the list and the list will be sent to the cloud.

To add a new operation to a child device create a new file in `/etc/tedge/operations/c8y/<child-device>` directory as below.

```shell
sudo -u tedge touch /etc/tedge/operations/c8y/<child-device>/c8y_Restart
```

> Now the new operation will be automatically added to the list of child supported operations and will be sent to the cloud.

Reviewer 2: Add a blank line as a separation from the previous paragraph.

```suggestion

Now the new operation will be automatically added to the list of child supported operations and will be sent to the cloud.
```

Author: fixed

Reviewer 2: That commit seems missing ;-)

Now the new operation will be automatically added to the list and the list will be sent to the cloud.

> To add a new operation to a child device create a new file in `/etc/tedge/operations/c8y/<child-device>` directory as below.

Reviewer: ```suggestion
To add a new operation to a child device, create a new file in `/etc/tedge/operations/c8y/<child-device>` directory as below.
```

Author: Fixed

```shell
sudo -u tedge touch /etc/tedge/operations/c8y/<child-device>/c8y_Restart
```

Now the new operation will be automatically added to the list of child supported operations and will be sent to the cloud.

### Removing supported operations

> To remove supported operation for a thin-edge device we can remove the file from `/etc/tedge/operations/c8y` directory. eg:

Reviewer: ```suggestion
To remove a supported operation for a thin-edge device, the corresponding operation file must be removed from the `/etc/tedge/operations/c8y` directory. eg:
```

Author: Fixed

```shell
sudo rm /etc/tedge/operations/c8y/c8y_Restart
```

> To remove supported operation for a child device we can remove the file from `/etc/tedge/operations/c8y/<child-device>` directory. eg:

Reviewer: ```suggestion
Similarly, the supported operation for a child device can be removed by removing the corresponding operation file from the child device operations directory at `/etc/tedge/operations/c8y/<child-device>`. Eg:
```

Author: Fixed
